mypy/fastparse.py

```diff
@@ -403,16 +403,30 @@ def visit_Delete(self, n: ast35.Delete) -> Node:
         else:
             return DelStmt(self.visit(n.targets[0]))
 
-    # Assign(expr* targets, expr value, string? type_comment)
+    # Assign(expr* targets, expr? value, string? type_comment, expr? annotation)
     @with_line
     def visit_Assign(self, n: ast35.Assign) -> Node:
         typ = None
-        if n.type_comment:
+        if hasattr(n, 'annotation') and n.annotation is not None:  # type: ignore
+            n_synt = True
+        else:
+            n_synt = False
+        if n_synt and self.pyversion < (3, 6):
+            raise TypeCommentParseError('Variable annotation syntax is only '
+                                        'suppoted in Python 3.6, use type '
+                                        'comment instead', n.lineno, n.col_offset)
+        if n.type_comment is not None:
```

Comment:
Could you add a comment noting that typed_ast prevents there from being both a type comment and an annotation?